.github/workflows/e2e-azure.yaml

```diff
@@ -6,6 +6,12 @@ on:
     - cron:  '0 6 * * *'
   push:
     branches: [ azure* ]
```

Comment:
Shouldn't the same be added here?

Reply:
i'm guessing this was added to make debugging easier and not have to trigger a run manually on every commit.
the changes i added were to make sure that if something changes in the `tests`  directory we run the azure tests since that directory contains the said tests.

Reply:
Yes, my point is that the same could be said for changes ending up in `main` that touch these things. In addition to not any branch starting with `azure` automatically trigger this, if the related files didn't change.